src/pages/library/RequestDatasetAccessModal.js

```diff
@@ -71,22 +72,26 @@ export const RequestDatasetAccessModal = ({ onDismiss, datasets }) => {
   ])
 }
 
-const RequestDatasetAccessButton = ({ dataset }) => {
+const RequestDatasetAccessButton = (dataset) => {
   const [requesting, setRequesting] = useState(false)
   const [requested, setRequested] = useState(false)
   const signal = Utils.useCancellation()
 
-  const requestAccess = _.flow(
-    Utils.withBusyState(setRequesting),
-    withErrorReporting('Error requesting dataset access')
-  )(async () => {
-    await Ajax(signal).DataRepo.requestAccess(dataset.id)
-    setRequested(true)
-  })
-
   return h(ButtonPrimary, {
     disabled: requesting || requested,
-    onClick: requestAccess
+    onClick: () => {
+      _.flow(
+        Utils.withBusyState(setRequesting),
+        withErrorReporting('Error requesting dataset access')
+      )(async () => {
+        await Ajax(signal).DataRepo.requestAccess(dataset.id)
```

Comment:
Does this need to be feature flagged?

Reply:
not sure what you mean @lu-c ?

Reply:
I think pete is asking, "Does it need a "is dataCatalog enabled" check before you run this request?"

The requestAccess endpoint doesnt exist ATM, so this request will just fail. That error is being caught and showing "error requesting dataset access" to the user. I don't think we need to feature flag this, esp since access to this page should be feature flagged, but if there is something we should do, let me know!